tests/v2/validation/nodescaling/auto_replace.go

```diff
@@ -0,0 +1,72 @@
+//go:build (validation || extended) && !infra.any && !infra.aks && !infra.eks && !infra.gke && !infra.rke2k3s && !cluster.any && !cluster.custom && !cluster.nodedriver && !sanity && !stress
+
+package nodescaling
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	shutdownCommand = "shutdown"
+	fleetNamespace  = "fleet-default"
+	deletingState   = "deleting"
+)
+
+type AutoReplaceSuite struct {
+	suite.Suite
+	client        *rancher.Client
+	session       *session.Session
+	rancherConfig *rancherversion.Config
+}
+
+func (s *AutoReplaceSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *AutoReplaceSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	rancherConfig := new(rancherversion.Config)
+	config.LoadConfig(rancherversion.ConfigurationFileKey, rancherConfig)
+	s.rancherConfig = rancherConfig
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+// ReplaceNodes replaces the last node with the specified role(s) in a k3s/rke2 cluster
+func (s *AutoReplaceSuite) TestAutoReplaceRKE2K3S(t *testing.T, client *rancher.Client, clusterName string, nodeToReplace *nodes.Node, machineName string) {
```

Comment:
Maybe this can be an offline discussion. But I think we should have static tests and a dynamic tests for this. Also, the structure of this test is a bit off to me.

This file should be called `auto_replace_test.go` and this function `TestAutoReplaceRKE2K3S` should be calling a single function to a separate file called `auto_replace.go`.